### PR TITLE
Update readme to better reflect what we have today

### PR DESCRIPTION
**Pull request by SgtCoDFish**
We got feedback that `make demo` doesn't work any more, so this is a quick refresh to improve the initial onboarding guide for anyone reading the README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,38 +11,44 @@
...
 
-The demo installation uses Helm, and roughly matches what you'd get by installing trust-manager into your own
-cluster using Helm - although it uses locally-built images rather than the ones we publish publicly.
+### Testing
 
-## Example Bundle
+trust-manager has various categories of tests. All categories are run against every PR, along with other checks.
+
+- `make test-unit` - Runs simpler, faster tests which test specific functions
+- `make test-integration` - Runs heavier tests with no Kubernetes cluster which test how different pieces work together
```
Comment (erikgb) [inline README.md:46]:
```suggestion
- `make test-integration` - Runs heavier tests with a simplified control-plane which tests how different pieces work together
```

File: README.md [hunk 2]
```diff
@@ -11,38 +11,44 @@
 
 trust-manager is the easiest way to manage trust bundles in Kubernetes and OpenShift clusters!
 
-It takes a list of trusted certificates which you specify and combines them into a bundle which you can trust directly
+It takes a list of trusted certificate sources and combines them into a bundle which you can trust directly
```
[the remark this answers is not in the document]
Comment (SgtCoDFish) [reply README.md:14]: I think we could go into more detail here but I think we'd be duplicating the docs. I'd rather not change this _too_ much, and instead direct people to read the website.

My feeling is that it'd be best to leave this as-is, and maybe revisit later. I'd prefer to quickly improve this and move on for now if that's OK?

[the remark this answers is not in the document]
Comment (erikgb) [reply README.md:14]: Sounds good!

**Review by erikgb — COMMENTED** [review]
/lgtm

🚀  Thanks!

**Pull request merged**